vendors/espressif/boards/esp32/ports/wifi/iot_wifi.c

```diff
@@ -263,24 +263,48 @@ BaseType_t WIFI_IsConnected( void )
 
 WIFIReturnCode_t WIFI_Off( void )
 {
+    esp_err_t ret;
```

Comment:
I see other APIs are guarded by a semaphore. Do we need a semaphore to prevent concurrent access?